bob/src/hw_metrics_collector.rs

```diff
@@ -133,14 +145,199 @@ impl HWMetricsCollector {
+                self.iostat_avl = false;
+            }
+        };
+        Err("iostat is not awailable".to_string())
+    }
+}
+
+struct DiskStatCollector {
+    iostat_avl: bool,
+    disk_metric_data: Vec<(String, String)>,
+}
+
+impl DiskStatCollector {
+    fn new(disks: &HashMap<PathBuf, String>) -> Self {
+        let mut disk_metric_data = Vec::new();
+        for (path, disk_name) in disks {
+            let path_str = path.as_os_str().to_str().unwrap();
+            if let Ok(dev_name) = Self::dev_name(path_str) {
+                let metric_prefix = format!("{}.{}", DISKS_FOLDER, disk_name);
+                disk_metric_data.push((metric_prefix, dev_name));
+            }
+        }
+
+        DiskStatCollector { 
+            iostat_avl: true,
+            disk_metric_data
+        }
+    }
+
+    fn collect_and_send_metrics(&mut self) {
+        for (metric_pref, dev_name) in &self.disk_metric_data {
+            if let Ok((iops, iowait)) = Self::collect_iostat(&mut self.iostat_avl, dev_name) {
+                let gauge_name = format!("{}_iowait", metric_pref);
+                gauge!(gauge_name, iowait);
+
+                let gauge_name = format!("{}_iops", metric_pref);
+                gauge!(gauge_name, iops);
+            }
+        }
+    }
+
+    fn dev_name(disk_path: &str) -> Result<String, String> {
+        let df = std::process::Command::new("df").arg(disk_path).output();
+        match df {
+            Ok(output) => {
+                if let (true, Ok(out)) = (output.status.success(), String::from_utf8(output.stdout))
+                {
+                    let mut lines = out.lines();
+                    lines.next(); // skip headers
+                    if let Some(line) = lines.next() {
+                        if let Some(raw_dev_name) = line.split_whitespace().next() {
+                            if let (Some(slash_ind), Some(non_digit_ind)) =
+                                (
+                                    // find where /dev/ ends
+                                    raw_dev_name.rfind('/'),
+                                    // find where partition digits start
+                                    raw_dev_name.rfind(|c: char| !c.is_digit(10)),
+                                )
+                            {
+                                return Ok(raw_dev_name[slash_ind + 1..=non_digit_ind].to_string());
+                            }
+                        }
+                    }
+                }
+            }
+            Err(e) => {
+                debug!("Failed to execute df: {}", e);
+                return Err("Failed to execute df".to_string());
+            }
+        };
+        Err("Error occured in dev_name getter".to_string())
+    }
+
+    fn parse_iostat_result(iostat_str: &str) -> Result<f64, std::num::ParseFloatError> {
+        iostat_str.replace(',', ".").parse()
+    }
+
+    fn collect_iostat(iostat_avl: &mut bool, disk_dev_name: &str) -> Result<(f64, f64), String> {
```

Comment:
`collect_iostat` is a part of `DiskStatCollector`, so it has direct access to `iostat_avl`, thus there is no need to pass it as a parameter